### ISSUE TITLE
Rotation leaves parts of pictures blank

### ISSUE THREAD

**Issue by SteveMan67** [darktable-org/darktable#15785]
### Describe the bug

When adjusting the Rotate and Perspective Rotation module, If adjusted to quickly, the program leaves this triangle of black at the edge of the picture, and it does not fix itself without setting the rotation back to zero again.

As you can see in the top right edge of this photo, it leaves just enough space to be noticeable, sometimes more.
![Screenshot 2023-11-29 205427](https://github.com/darktable-org/darktable/assets/143290011/ce592448-b3f1-4caa-86bb-639e65e63215)

### Steps to reproduce

1. Open Darkroom
2. Go to the rotate and perspective module and adjust the rotation

### Expected behavior

crop as it is rotating to eliminate blank space.

### Logfile | Screenshot | Screencast

![Screenshot 2023-11-29 205427](https://github.com/darktable-org/darktable/assets/143290011/7ce57203-af3f-4d84-93e7-57d12d0b667d)

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.4.2

### What OS are you using?

Windows

### What is the version of your OS?

Windows 10

### Describe your system?

**Processer**    __Intel(R) Xeon(R) CPU E31225 @ 3.10GHz   3.10 GHz__
**Installed RAM** 20.0 GB
**System Type**    64-bit operating system, x64-based processor

### Are you using OpenCL GPU in darktable?

I dont know

**Comment by wpferguson** [comment]
What do you have automatic cropping set to?

**Comment by elstoc** [comment]
It doesn't matter -- in some cases (at least in 4.4.2) it fails to crop properly regardless of the setting. This may have been fixed in master but I haven't been able to check since I haven't found a consistent reproducer.

**Comment by wpferguson** [comment]
I may have seen this happen a long time ago, but I haven't seen it in the last couple of years and I use it a lot (several hundred images per week) and I'm OCD about verticals being vertical.  I'll see if I can reproduce on windows.

**Comment by wpferguson** [comment]
Tried on windows with and without lens correction, in case there was an interaction.  Unable to reproduce.  I don't have opencl on the VM, so it was a CPU only test.

**Comment by jenshannoschwalm** [comment]
There were interpolators fixes done in master. This could also be affected.

**Comment by elstoc** [comment]
For me it sometimes just looks like it's ignored the contents of the combobox and chosen not to crop anyway.

**Comment by ralfbrown** [comment]
I've seen it on occasion when doing fast repeated small rotations via keyboard shortcut.  The most recent time this occurred, I was able to fix it by switching automatic cropping to "largest area" and back to "original format".  So it's probably a race condition.

**Comment by jenshannoschwalm** [comment]
> I've seen it on occasion when doing fast repeated small rotations via keyboard shortcut.

Yes i have also seen that but **only** while using keyboard shortcuts. Don't understand ...

**Comment by jenshannoschwalm** [comment]
@ralfbrown can you confirm that while using shortcuts with key combinations like ctrl_r there is some issue dependency on the Ctrl key still being down?

**Comment by elstoc** [comment]
I've seen it when going back to earlier points in the history stack as well.

**Comment by ralfbrown** [comment]
I use Alt-[ and Alt-] as my rotation shortcuts.  I do usually leave Alt pressed when doing rotations until I'm sure I'm done.

**Comment by Donatzsky** [comment]
Just had it happen to me. Used right-click drag to straighten a line, used ctrl-z to undo. 
When exporting it also looks like that.

![image](https://github.com/darktable-org/darktable/assets/4938390/78fabaaa-c312-46bc-99bc-786d4d8df23c)

Rotate and perspective is still active after undoing:
![image](https://github.com/darktable-org/darktable/assets/4938390/89917132-7309-42d2-bc02-76134a6f9293)

History stack:
![image](https://github.com/darktable-org/darktable/assets/4938390/e84792d5-af9e-49f4-835d-8729fa3c843c)
The local contrast was just to see if doing something else would fix it.

It's almost as if ctrl-z only undoes the crop, not the rotate.

Not using OpenCL.
